src/main/java/seedu/address/model/book/BookStatus.java

```diff
@@ -0,0 +1,172 @@
+package seedu.address.model.book;
+
+import static seedu.address.commons.util.AppUtil.checkArgument;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+import static seedu.address.model.book.BookStatusType.AVAILABLE;
+import static seedu.address.model.book.BookStatusType.BORROWED;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Optional;
+
+import seedu.address.model.patron.Patron;
+
+/**
+ * Represents a BookStatus in LibTask.
+ * Guarantees: immutable; is valid as declared in {@link #isValidStatus(BookStatusType, Optional, Optional, Optional)}
+ */
+public class BookStatus {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "An available book status must not contain any borrower, borrowDate and return date "
+                    + "and a borrowed status must contain borrower, borrowDate and return date,"
+                    + "and date must be in dd-MMM-yyyy format.";
+    public static final SimpleDateFormat STATUS_DATE_FORMAT = new SimpleDateFormat("dd-MMM-yyyy");
+    public static final String VALIDATION_REGEX = "^\\d{2}-[a-zA-Z]{3}-\\d{4}$";
```

Comment:
Tested the regex, works fine as intended.

But some MMM are accepted such as "out", "abc" and "xyz". I think this is still acceptable as there are date validation in ```parseToDate``` and ```isParseable```. Good job on that :100:


![image](https://user-images.githubusercontent.com/39936106/158505564-eb27ed7f-6c88-4fa0-af91-c5ad109e4e0a.png)